lib/matplotlib/figure.py

```diff
@@ -1025,48 +1025,50 @@ def subplots(self, nrows=1, ncols=1, sharex=False, sharey=False,
+            - If False, no axis will be shared amongst subplots.
+            - If a string must be one of "row", "col", "all", or "none".
+                - "all" has the same effect as True.
+                - "none" has the same effect as False.
+                - If "row", each subplot row will share a x-axis.
+                - If "col", each subplot column will share a x-axis.
+
+            Note that if the x-axis is shared across rows (sharex=True or
+            sharex="col"), then the x tick labels will only be display on
+            subplots of the bottom row.
 
         sharey : {"none", "all", "row", "col"} or bool, default: False
-            If *False*, or "none", each subplot has its own Y axis.
-
-            If *True*, or "all", all subplots will share an Y axis, and the y
-            tick labels on all but the first column of plots will be invisible.
-
-            If "row", each subplot row will share an Y axis, and the y tick
-            labels on all but the first column of plots will be invisible.
-
-            If "col", each subplot column will share an Y axis.
+            - If True, the y-axis will be shared amongst all subplots.
+            - If False, no y-axis will be shared amongst subplots.
+            - If a string must be one of "row", "col", "all", or "none".
+                - "all" has the same effect as True.
+                - "none" has the same effect as False.
+                - If "row", each subplot row will share a y-axis.
+                - If "col", each subplot column will share a y-axis
+
+            Note that if the y-axis is shared across columns (sharey=False or
+            sharey="col"), 1 then the y tick labels will only be displayed on
+            subplots of the first column.
 
         squeeze : bool, default: True
-            If *True*, extra dimensions are squeezed out from the returned axes
-            array:
-
-            - if only one subplot is constructed (nrows=ncols=1), the resulting
-              single Axes object is returned as a scalar.
-
-            - for Nx1 or 1xN subplots, the returned object is a 1-d numpy
-              object array of Axes objects are returned as numpy 1-d arrays.
-
-            - for NxM subplots with N>1 and M>1 are returned as a 2d array.
-
-            If *False*, no squeezing at all is done: the returned object is
-            always a 2-d array of Axes instances, even if it ends up being 1x1.
+            - If True, extra dimensions are squeezed out from the returned
+              axis object:
+
+                - if only one subplot is constructed (nrows=ncols=1), the
+                  resulting single Axis object is returned as a scalar.
+                - for Nx1 or 1xN subplots, the returned object is a 1-d numpy
+                  object array of Axis objects are returned as numpy 1-d
+                  arrays.
+                - for NxM subplots with N>1 and M>1 are returned as a 2d
+                  arrays.
```

Comment:
* '2-d array'? 
* Besides, I feel like some subject is missing (but my English may be wrong). I would have expected something like: 'for NxM subplots with N>1 and M>1, the returned object is a 2-d numpy object array of Axes objects.'